cdk/lib/db.ts

```diff
@@ -54,58 +54,60 @@ interface ICacclDb {
 }
 
 export abstract class CacclDbBase extends Construct implements ICacclDb {
-
   host: string;
+
   port: string;
-  dbPasswordSecret: Secret;
+
+  dbPasswordSecret: secretsmanager.Secret;
+
   clusterParameterGroupParams: { [key: string]: string } = {};
+
   instanceParameterGroupParams: { [key: string]: string } = {};
-  dbCluster: DocDbDatabaseCluster | RdsDatabaseCluster;
+
+  dbCluster: docdb.DatabaseCluster | rds.DatabaseCluster;
+
   metricNamespace: string;
-  metrics: { [key: string]: Metric[] };
-  alarms: Alarm[];
+
+  metrics: { [key: string]: cloudwatch.Metric[] };
+
+  alarms: cloudwatch.Alarm[];
+
   removalPolicy: RemovalPolicy;
-  dbSg: SecurityGroup;
+
+  dbSg: ec2.SecurityGroup;
 
   // the "etcetera" policy for the parameter group(s) and security group
   etcRemovalPolicy: RemovalPolicy;
 
-  static createDbConstruct(scope: Construct, props: CacclDbProps) {
-    const { options } = props;
-    switch (options.engine.toLowerCase()) {
-      case 'docdb':
-        return new CacclDocDb(scope, 'DocDb', props);
-      case 'mysql':
-        return new CacclRdsDb(scope, 'RdsDb', props);
-      default:
-        throw Error(`Invalid dbOptions.engine value: ${options.engine}`);
-    }
-  }
-
   constructor(scope: Construct, id: string, props: CacclDbProps) {
     super(scope, id);
 
     const { vpc } = props;
-    const {
-      removalPolicy = DEFAULT_REMOVAL_POLICY,
-    } = props.options;
+    const { removalPolicy = DEFAULT_REMOVAL_POLICY } = props.options;
 
     // removal policy for the cluster & instances
     this.removalPolicy = (<any>RemovalPolicy)[removalPolicy];
     // if we're keeping the dbs we have to keep the parameter group
     // and security group or cloudformation will barf. Also there's no
     // "SNAPSHOT" option for these resources so it's either "RETAIN" or "DESTROY"
-    this.etcRemovalPolicy = this.removalPolicy === RemovalPolicy.RETAIN
-      ? RemovalPolicy.RETAIN
-      : RemovalPolicy.DESTROY;
-
-    this.dbPasswordSecret = new Secret(this, 'DbPasswordSecret', {
-      description: `docdb master user password for ${Stack.of(this).stackName}`,
-      generateSecretString: {
-        passwordLength: 16,
-        excludePunctuation: true,
+    this.etcRemovalPolicy =
+      this.removalPolicy === RemovalPolicy.RETAIN
+        ? RemovalPolicy.RETAIN
+        : RemovalPolicy.DESTROY;
+
+    this.dbPasswordSecret = new secretsmanager.Secret(
+      this,
+      'DbPasswordSecret',
+      {
+        description: `docdb master user password for ${
+          Stack.of(this).stackName
```

Comment:
Not sure what this needs to be on its own line, but I don't care that much. it's still readable :)